### ISSUE TITLE
Problems with downloading Exported Course

### ISSUE THREAD

**Issue by frohro** [regisb/openedx-docker#74]
I stopped apache2.  I did a git pull.  I changed the ports to 80:80 and 443:443 in docker-compose.yml.  I did a make configure with fweb.wallawalla.edu and studio.fweb.wallawalla.edu, and enabled student notes and portaniner, and did a git update.  Then I exported my test course.  It finished making the tar.gz file, but trying to download it gives me an error in the browser, with an apology and in the logs I get this.  I disabled my ad blockers in my browser without any effect.  Here are the errors:

cms_1            | 2018-10-02 00:13:56,231 INFO 12 [tracking] logger.py:50 - {"username": "frohro", "event_type": "/export_output/course-v1:wallawalla+test101+fall2018", "ip": "172.20.18.73", "agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36", "host": "studio.fweb.wallawalla.edu", "referer": "http://studio.fweb.wallawalla.edu/export/course-v1:wallawalla+test101+fall2018", "accept_language": "en-US,en;q=0.9", "event": "{\"POST\": {}, \"GET\": {}}", "event_source": "server", "context": {"user_id": 2, "org_id": "", "course_id": "", "path": "/export_output/course-v1:wallawalla+test101+fall2018"}, "time": "2018-10-02T00:13:56.230915+00:00", "page": null}
cms_1            | 2018-10-02 00:13:56,245 ERROR 12 [root] signals.py:17 - Uncaught exception from None
cms_1            | Traceback (most recent call last):
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/exception.py", line 41, in inner
cms_1            |     response = get_response(request)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
cms_1            |     response = self._get_response(request)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 187, in _get_response
cms_1            |     response = self.process_exception_by_middleware(e, request)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 185, in _get_response
cms_1            |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/views/decorators/http.py", line 40, in inner
cms_1            |     return func(request, *args, **kwargs)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/utils/decorators.py", line 149, in _wrapped_view
cms_1            |     response = view_func(request, *args, **kwargs)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/contrib/auth/decorators.py", line 23, in _wrapped_view
cms_1            |     return view_func(request, *args, **kwargs)
cms_1            |   File "/openedx/edx-platform/common/djangoapps/util/views.py", line 50, in inner
cms_1            |     response = view_func(request, *args, **kwargs)
cms_1            |   File "/openedx/edx-platform/cms/djangoapps/contentstore/views/import_export.py", line 427, in export_output_handler
cms_1            |     tarball = course_import_export_storage.open(artifact.file.name)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/core/files/storage.py", line 38, in open
cms_1            |     return self._open(name, mode)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/core/files/storage.py", line 300, in _open
cms_1            |     return File(open(self.path(name), mode))
cms_1            | IOError: [Errno 2] No such file or directory: u'/openedx/data/uploads/user_tasks/2018/10/02/course.PvxqpC.tar.gz'
cms_1            | 2018-10-02 00:13:56,246 ERROR 12 [django.request] exception.py:135 - Internal Server Error: /export_output/course-v1:wallawalla+test101+fall2018
cms_1            | Traceback (most recent call last):
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/exception.py", line 41, in inner
cms_1            |     response = get_response(request)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
cms_1            |     response = self._get_response(request)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 187, in _get_response
cms_1            |     response = self.process_exception_by_middleware(e, request)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 185, in _get_response
cms_1            |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/views/decorators/http.py", line 40, in inner
cms_1            |     return func(request, *args, **kwargs)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/utils/decorators.py", line 149, in _wrapped_view
cms_1            |     response = view_func(request, *args, **kwargs)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/contrib/auth/decorators.py", line 23, in _wrapped_view
cms_1            |     return view_func(request, *args, **kwargs)
cms_1            |   File "/openedx/edx-platform/common/djangoapps/util/views.py", line 50, in inner
cms_1            |     response = view_func(request, *args, **kwargs)
cms_1            |   File "/openedx/edx-platform/cms/djangoapps/contentstore/views/import_export.py", line 427, in export_output_handler
cms_1            |     tarball = course_import_export_storage.open(artifact.file.name)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/core/files/storage.py", line 38, in open
cms_1            |     return self._open(name, mode)
cms_1            |   File "/usr/local/lib/python2.7/dist-packages/django/core/files/storage.py", line 300, in _open
cms_1            |     return File(open(self.path(name), mode))
cms_1            | IOError: [Errno 2] No such file or directory: u'/openedx/data/uploads/user_tasks/2018/10/02/course.PvxqpC.tar.gz'
nginx_1          | 172.20.18.73 - - [02/Oct/2018:00:13:56 +0000] "GET /export_output/course-v1:wallawalla+test101+fall2018 HTTP/1.1" 500 8194 "http://studio.fweb.wallawalla.edu/export/course-v1:wallawalla+test101+fall2018" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36" "-"
Here is my make info:

me@fweb:~/openedx-docker$ make info
uname -a
Linux fweb 4.15.0-34-generic #37-Ubuntu SMP Mon Aug 27 15:21:48 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
-------------------------
git rev-parse HEAD
b4947d10e7106e48d0bf5b59bdcec1fcc14382ec
-------------------------
docker version
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:24:51 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:23:15 2018
  OS/Arch:          linux/amd64
  Experimental:     false
-------------------------
docker-compose --version
docker-compose version 1.21.2, build a133471
-------------------------
echo $EDX_PLATFORM_PATH

echo $EDX_PLATFORM_SETTINGS

This is not really important to me yet, but may be someday.  :-)  Now, back to trying to import a course. 

Thanks!
Rob

**Comment by frohro** [comment]
I did a docker purge -a and re-installed everything without apache2 running.  I did the import test, and had the same failure I got before.  ("Unpacking error importing course" in the browser.) I then tried to export a test course to see if it would import something it just exported, and I ran into the same export problem described here.  Here is a log of the cms in case it helps.
[log_cms_after_failed_import.zip](https://github.com/regisb/openedx-docker/files/2436391/log_cms_after_failed_import.zip)

**Comment by regisb** [comment]
@frohro I managed to reproduce your problem. It was due to an incorrect file sharing setup between the CMS and CMS workers, and should now be fixed. Just run `git pull` and `make stop`, then `make run` again and you should be fine.

**Comment by frohro** [comment]
Thank you!  I was able to import my course using studio.fweb.wallawalla.edu with apache2 off.   I tried to switch to forwarding ports as I was, but the whole course disappeared.  When I tried to create it again, it said it was there.  I switched back to without apache2 and the regular ports, did a make reindex-courses, but it still didn't appear.  I'm building from scratch again and am going to try starting with apache2 and port forwarding.  

I really appreciate how fast you fix bugs.  Thanks again.
